Source/SwiftLintFramework/Rules/Lint/DuplicatedKeyInDictionaryLiteralRule.swift

```diff
@@ -117,8 +132,36 @@ public struct DuplicatedKeyInDictionaryLiteralRule: ASTRule, ConfigurationProvid
         }
     }
 
+    private func isCodeGeneratedKey(keyExpression: String) -> Bool {
+      let contentFile = File(contents: keyExpression)
+      let request = Request.editorOpen(file: contentFile)
+
+      guard let response = try? request.send() else { return false }
```

Comment:
I'm not absolutely sure, but another SourceKit request might be expensive. It doesn't provide much value either since we can already see from the string whether it is `#line` (the simplest check) or is a call (contains parentheses). Thus, another request is not actually needed here.

Reply:
OK, note 2 things:
1.  Checking the `kind` is more general and also contains other directives like `#sourceLocation` and any other directive that will be added to the language in the future.
2. `SwiftExpressionKind.call` probably contains other scenarios that I don't think about (see new implementation)

Reply:
Valid points. The cases we know of are covered now while still being very fast. If more scenarios come up or the language is extended we can enhance the implementation.

Another approach could be an implementation based on the real AST using SwiftSyntax. It is fast and as exact as we can be without a full compilation.